src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -32,6 +31,116 @@ internal void VerifyDiagnostics(string source, CSharpCompilationOptions options,
     public class LocalFunctionTests : LocalFunctionsTestBase
     {
         [Fact]
+        public void LocalFunctionAttributeOnTypeParameter()
+        {
+            var tree = SyntaxFactory.ParseSyntaxTree(@"
+using System;
+class C
+{
+    public void M()
+    {
+        void Local<[A, B, CLSCompliant, D]T>() { }
```

Comment:
No, we should test SemanticModel for the syntax that is part of the attribute application. The APIs that we should test are: GetTypeInfo, GetSymbolInfo, GetAliasInfo, Lookup...